### PR TITLE
Update to latest release (`0.6.0`) for `embassy-executor` in `esp-embassy-hal` (fixes #1941)

### LINKED ISSUE #1941: Update `embassy-executor` dependency in esp-hal-embassy crate.

### PR DESCRIPTION
**Pull request by Sycrosity**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

#### Description

Fixes #1941 by updating the embassy-executor crate to the latest version in the `esp-embassy-hal` crate, and the `hil-test` and `examples` crates.

#### Testing

I tested a number of examples (namely: wifi_access_point, wifi_embassy_access_point, embassy_hello_world) on an esp32c3, which all functioned as expected.

I also tested my own program which uses a variety of esp-hal projects, which led to me finding the `OneShotTimer` problem.

### CODE REVIEW

File: esp-hal-embassy/src/time_driver.rs [hunk 1]
```diff
@@ -12,7 +12,7 @@ use esp_hal::{
 
 pub const MAX_SUPPORTED_ALARM_COUNT: usize = 7;
 
-pub type Timer = OneShotTimer<'static, ErasedTimer>;
+pub type Timer = OneShotTimer<ErasedTimer>;
```
Comment (bugadani) [inline esp-hal-embassy/src/time_driver.rs:15]: Please undo this change

Comment (Sycrosity) [reply esp-hal-embassy/src/time_driver.rs:15]: fair point, completely missed that - whats the best way to pin versions while testing? would it be to fork from the v0.19.0 tag or just check these things manually?

**Review by bjoernQ — APPROVED** [review]
LGTM
